### PR TITLE
Add Puppet 4 parameter validation and add some fixes for test

### PR DESCRIPTION
**Pull request by baurmatt**
Hey,

I've added Puppet 4 parameter validation and got rid of the deprecated validate_* function. While digging through the code I come over somethings which aren't considers best practices anymore and changed them.

I think the easiest way to review this change is commit by commit :)

Thanks! :)

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
   "description": "Puppet module to manage ProFTPD on FreeBSD and Linux with very flexible, hiera-friendly configuration",
   "dependencies": [
     { "name": "puppetlabs/concat", "version_requirement": ">= 1.0.0 <5.0.0" },
-    { "name": "puppetlabs/stdlib", "version_requirement": ">= 4.2.0 <5.0.0" }
+    { "name": "puppetlabs/stdlib", "version_requirement": ">= 4.25.0 <5.0.0" }
```
Comment (fraenki) [inline metadata.json]: What's the reason for this version bump?
I think 4.6.0 would be sufficient to support `assert_private()`.

Comment (baurmatt) [reply metadata.json]: The reason is, that `Stdlib::Filesource` is only available in >= 4.25.0

Comment (fraenki) [reply metadata.json]: Thanks for the clarification!

**Comment by fraenki** [comment]
@baurmatt Merged, thanks! This will be included in the upcoming release 1.5.0.

**Pull request merged**